phpBB/phpbb/storage/file_info.php

```diff
@@ -62,6 +81,25 @@ public function __construct(adapter_interface $adapter, $path)
 	 */
 	public function get($name)
 	{
+		// Try to read properties from db if isn't set
+		if (!isset($this->properties[$name]))
+		{
+			$sql = 'SELECT metadata
+				FROM ' .  $this->storage_table . "
+				WHERE storage = '" . $this->db->sql_escape($this->storage_name) . "'
+					AND file_path = '" . $this->db->sql_escape($this->path) . "'";
+			$result = $this->db->sql_query($sql);
+
+			$metadata =  json_decode($this->db->sql_fetchfield('metadata'), true);
+
+			if ($metadata !== null)
+			{
+				$this->properties = $metadata;
+			}
+
+			$this->db->sql_freeresult($result);
+		}
+
 		if (!isset($this->properties[$name]))
```

Comment:
Is there any reason why you have the exact same if twice?

Reply:
Yes, because it can be defined in the first condition.
Yes, i can probably merge it